editors/document-model/editor-schema.tsx

```diff
@@ -68,13 +73,26 @@ export default function EditorSchema({
     ...props
 }: IProps) {
     const editorRef = useRef<editor.IStandaloneCodeEditor | null>(null);
+    const constrainedEditorRef = useRef<ConstrainedInstance | null>(null);
+    const modelEditorRef = useRef<editor.ITextModel | null>(null);
     const [code, setCode] = useState(value || '');
 
     useEffect(() => {
         const normalizedSchema = normalizeSchema(value, scope, name);
 
         if (normalizedSchema !== code) {
+            constrainedEditorRef.current?.removeRestrictionsIn(
+                modelEditorRef.current,
+            );
+
             setCode(normalizedSchema);
+
+            setTimeout(() => {
```

Comment:
Due code changes are not reflected instantly in monaco editor, I need to delay a little bit the re-apply restrictions rules